src/force-app/main/default/classes/tests/TestCasesService.cls

```diff
@@ -132,4 +132,85 @@ private class TestCasesService {
+        User inactiveUser = createUser(false);
+        insert inactiveUser;
+
+        Id caseId = fflib_IDGenerator.generate(Case.SObjectType);
+        Case oldCase = new Case(
+            Id = caseId,
+            OwnerId = inactiveUser.Id,
+            Status = Constants.CASE_STATUS_NEW
+        );
+        Case newCase = new Case(
+            Id = caseId,
+            OwnerId = inactiveUser.Id,
+            Status = Constants.CASE_STATUS_CLOSED
+        );
+        Map<Id, Case> oldRecords = new Map<Id, Case> { caseId => oldCase };
+        List<Case> newRecords = new List<Case>{ newCase };
+        
+        CasesService.getInstance().preventStatusChangeByNotAnOwner(newRecords, oldRecords);
+        System.assert(!newCase.hasErrors());
+    }
+
+    @isTest
+    private static void testPreventStatusChangeByNotAnOwner_InactiveOwner_Negative() {
+        User activeUser = createUser(true);
+        insert activeUser;
+
+        Id caseId = fflib_IDGenerator.generate(Case.SObjectType);
+        Case oldCase = new Case(
+            Id = caseId,
+            OwnerId = activeUser.Id,
+            Status = Constants.CASE_STATUS_NEW
+        );
+        Case newCase = new Case(
+            Id = caseId,
+            OwnerId = activeUser.Id,
+            Status = Constants.CASE_STATUS_CLOSED
+        );
+        Map<Id, Case> oldRecords = new Map<Id, Case> { caseId => oldCase };
+        List<Case> newRecords = new List<Case>{ newCase };
+        
+        CasesService.getInstance().preventStatusChangeByNotAnOwner(newRecords, oldRecords);
+        System.assert(newCase.hasErrors(), Constants.ERROR_RECORD_SHOULD_BE_ERRORED);
+        List<Database.Error> errors = newCase.getErrors();
+        System.assertEquals(
+            Constants.ERROR_PLEASE_CLAIM_CASE_BEFORE_CHANGING_ITS_STATUS,
+            errors[0].getMessage(),
+            Constants.ERROR_RECORD_HAS_UNEXPECTED_ERROR
+        );
+    }
+
+    @isTest
+    private static void testGetSobjectTypeById_Positive() {
+        Account acc = new Account(Name='test Account');
+        insert acc;
+        String accName = CasesService.getSobjectTypeById(acc.Id);
+        System.assertEquals('Account', accName, 'SObject Name does not match');
+    }
+
+    private static User createUser(Boolean isActive) {
+        Id adminProfileId = getAdminProfileId();
+        return new User(
+            FirstName = 'Test',
+            LastName = 'Testenko',
+            UserName = 'preventTest@test.com',
+            Email = 'preventTest@test.org',
+            Alias = 'ttest',
+            IsActive = isActive,
+            ProfileId = adminProfileId,
+            LanguageLocaleKey = 'en_US',
+            EmailEncodingKey = 'UTF-8',
+            LocaleSidKey = 'en_US',
+            TimeZoneSidKey = 'America/Los_Angeles'
+        );
+    }
+
+    private static Id getAdminProfileId() {
+        return [SELECT Id FROM Profile WHERE Name = :Constants.ADMIN_PROFILE_NAME LIMIT 1][0].Id;
```

Comment:
This could be simplified to `return [SELECT Id FROM Profile WHERE Name = :Constants.ADMIN_PROFILE_NAME LIMIT 1].Id;`